app/src/ble.c

```diff
@@ -222,6 +222,23 @@ int zmk_ble_clear_bonds() {
     return 0;
 };
 
+int zmk_ble_clear_all_bonds() {
+    LOG_DBG("zmk_ble_clear_all_bonds()");
+
+    // Unpair all profiles
+    for (uint8_t i = 0; i < ZMK_BLE_PROFILE_COUNT; i++) {
+        if (bt_addr_le_cmp(&profiles[i].peer, BT_ADDR_LE_ANY)) {
+            bt_unpair(BT_ID_DEFAULT, &profiles[i].peer);
+            set_profile_address(i, BT_ADDR_LE_ANY);
+        }
+    }
+
+    // Automatically switch to profile 0
+    zmk_ble_prof_select(0);
+
+    return 0;
```

Comment:
If this function doesn't have any errors to report, then it shouldn't return anything. Having a return code means that callers of this function will likely be written like
```c
int rc = zmk_ble_clear_all_bonds();
if (rc) {
  // This is dead code
  LOG_ERR("Failed to clear all bonds");
}
```

Alternatively, `bt_unpair()` *does* have a return code which we are currently ignoring. If we changed this function to report errors from the functions it calls, then the return code would be meaningful.

(This appears to also be an issue with multiple other functions in this file too.)